source/jormungandr/jormungandr/scenarios/helper_classes/streetnetwork_path.py

```diff
@@ -223,15 +224,27 @@ def _async_request(self):
             for destination in self.get_pt_objects(self._dest_obj):
                 self._futures.append(self._future_manager.create_future(self._do_request, origin, destination))
 
-    def wait_and_get(self):
+    def wait_and_get(self, timeout=None):
 
-        best_res = min(
-            (future.wait_and_get() for future in self._futures if future.wait_and_get().response is not None),
-            key=lambda r: r.response.journeys[0].duration,
-            default=None,
-        )
+        # timeout=None -> wait forever...
+        timer = gevent.timeout.Timeout(timeout, exception=False)
+
+        best_res = None
+        with timer:
+            best_res = min(
+                (
+                    future.wait_and_get()
+                    for future in self._futures
+                    if future.wait_and_get().response is not None
```

Comment:
If I understand correctly, the timeout will be for the completion of all futures in self._futures, and not on each individual future. 
Is this what you want to do ?

Reply:
yes! I cannot determine the best journey until all requests are completed

Reply:
Ok, I understand that you need the completion of all futures to determine the best journey.
But is it a good idea to put the timer on the completion of all futures, and not on each individual future ?
If you have a lot of futures, maybe 100ms will not be enough to compute all the futures ?